Firestore/tests/System/DocumentAndCollectionTest.php

```diff
@@ -196,4 +196,65 @@ public function testRootCollections()
             iterator_to_array($client->collections())
         );
     }
+
+    public function testCollectionsWithReadTime()
+    {
+        $childName = uniqid(self::COLLECTION_NAME);
+        $child = $this->document->collection($childName);
+        self::$localDeletionQueue->add($child);
+        $child->add(['name' => 'John']);
+        // without sleep, emulator system test may fail intermittently
+        sleep(1);
+
+        $readTime = new Timestamp(new \DateTimeImmutable());
+        $collection = $this->document->collections([
+            'readTime' => $readTime
+        ])->current();
+
+        $this->assertEquals($childName, $collection->id());
+    }
+
+    public function testRootCollectionsWithReadTime()
+    {
+        // ListCollectionIds request doesn't support read_time in options
+        // in emulator, thus skipping the tests for now.
+        $this->skipEmulatorTests();
```

Comment:
probably it supports now, @vishwarajanand to check

Reply:
I've added a task to change this in internal taskboard. We can merge it now.

Reply:
sure.. just pasting the link here for easy ref: [b/269205767](http://b/269205767)

Reply:
Are we ready to remove this?

Reply:
@yash30201 once it gets merged, pls also upgrade the emulator version [here](https://github.com/googleapis/google-cloud-php/blob/8ccc8c1b9ebb0cf199c86a0bad4eb13c1a9c9fb9/.github/workflows/firestore-emulator-system-tests.yaml#L22) in a separate PR.

Reply:
The emulator change is merged. The PR for emulator version bump is this: #6017